webapp/pages/profile/_id/_slug.vue

```diff
@@ -199,18 +199,16 @@
           </template>
           <template v-else>
             <p style="text-align: center; opacity: .5;">
-              niemand folgt {{ user.name }}
+              niemand folgt {{ userName }}
             </p>
           </template>
         </ds-card>
-        <ds-space 
+        <ds-space
           v-if="user.socialMedia && user.socialMedia.length"
           margin="large"
         >
           <ds-card style="position: relative; height: auto;">
-            <ds-space
-              margin="x-small"
```

Comment:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-04-17T14:07:43Z" title="Wednesday, April 17th 2019, 4:07:43 pm +02:00">Apr 17, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -430,6 +430,14 @@ export default {
         },
         fetchPolicy: 'cache-and-network'
       })
+    },
+    userName(maxLength) {
+      // Return Anonymous if no Username is given
+      if (!this.user.name) {
```


```suggestion
  return this.user.name || this.$t('profile.userAnonym')
```

Reply:
> [<img alt="ulfgebhardt" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/ulfgebhardt) **Authored by [ulfgebhardt](https://github.com/ulfgebhardt)**
_<time datetime="2019-04-19T09:57:40Z" title="Friday, April 19th 2019, 11:57:40 am +02:00">Apr 19, 2019</time>_

---


codewise not recomended since we have a possible truncate on user.name. Therefore it is wise to check if user.name is present unless we want to have `|| this.$t('profile.userAnonym')` duplicated and a very complicated statement.

```
export default {
  methods: {
    userName(userName, maxLength) {
      // Return Anonymous if no Username is given
      if (!userName) {
        return this.$t('profile.userAnonym')
      }
      // Return full Username or truncated Username
      return maxLength ? this.$filters.truncate(userName, maxLength) : userName
    }
  }
}
```

Suggestion: no triage

Reply:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-04-19T11:47:10Z" title="Friday, April 19th 2019, 1:47:10 pm +02:00">Apr 19, 2019</time>_

---


? :thinking: if you truncate the `userName` the value cannot be falsy.

My suggestion: In case you are not so sure if a situation can ever happen, write a test for it and see! :+1:

Reply:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-04-19T11:49:17Z" title="Friday, April 19th 2019, 1:49:17 pm +02:00">Apr 19, 2019</time>_

---


Ahh now I get it. Yes, the best is a guard clause, like you wrote it. Just put it on one line:
```js
if (!userName) return this.$t('profile.userAnonym')
```

Reply:
> [<img alt="ulfgebhardt" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/ulfgebhardt) **Authored by [ulfgebhardt](https://github.com/ulfgebhardt)**
_<time datetime="2019-04-20T11:33:01Z" title="Saturday, April 20th 2019, 1:33:01 pm +02:00">Apr 20, 2019</time>_

---


That's not lint compatible (most likely). But I will exclude the truncate and use it externally. I will refactor this and separate the two functionalities